### ISSUE TITLE
[ResourceList] Undesirable border in "more filters" modal on small screens

### ISSUE THREAD

**Issue by kaelig** [Shopify/polaris-react#2264]
## Issue summary

In the example "Resource List with all of its elements", when opening the modal and expanding the accordion, a border shows up where it shouldn't.

## Expected behavior

![Screen_Shot_2019-10-08_at_11_54_05_AM](https://user-images.githubusercontent.com/85783/66424395-693c5680-e9c2-11e9-8bf4-a1f4dbc21726.png)

## Actual behavior

![Screen_Shot_2019-10-08_at_11_49_51_AM](https://user-images.githubusercontent.com/85783/66424451-853ff800-e9c2-11e9-82b5-1b7ccba4a9e4.png)

## Steps to reproduce the problem

1. Go to https://polaris-react.herokuapp.com/?path=/story/all-components-resource-list--resource-list-with-all-of-its-elements
2. Click **More filters**
3. Expand **Tagged with**

**Comment by dleroux** [comment]
Looks like on my phone (ios) there’s also an extra space. ![5A105210-D7AF-478C-B46F-68DE87EFD69A](https://user-images.githubusercontent.com/1229901/66479493-9e9c7f00-ea6a-11e9-8479-999b7c2dd34f.jpeg)

**Comment by shui91** [comment]
I'd like to fix this border issue if someone could assign it to me (not sure if it needs to be or not!). 

The issue seems to appear even on my desktop view in the Storybook example.

<img width="1439" alt="Screen Shot 2019-10-13 at 11 26 35 PM" src="https://user-images.githubusercontent.com/11592023/66732486-fd6a4b80-ee10-11e9-8c1c-0f508ac53704.png">

**Comment by shui91** [comment]
> Looks like on my phone (ios) there’s also an extra space. ![5A105210-D7AF-478C-B46F-68DE87EFD69A](https://user-images.githubusercontent.com/1229901/66479493-9e9c7f00-ea6a-11e9-8479-999b7c2dd34f.jpeg)

@kaelig  @dleroux  I tested this on Chrome and Firefox and it seems like this is a Safari specific issue where there is 2px of margin is applied to the top and bottom of the Filter Trigger button.

I didn't fix it in #2284 as it seems out of scope for this issue.

<img width="518" alt="Screen Shot 2019-10-14 at 1 47 01 AM" src="https://user-images.githubusercontent.com/11592023/66738980-948cce80-ee24-11e9-989a-2a56d8fba165.png">

**Comment by kaelig** [comment]
There's still an extra space in iOS after #2284 was merged, I'll open a separate issue.

<img width="584" alt="Screen Shot 2019-10-14 at 11 57 09 AM" src="https://user-images.githubusercontent.com/85783/66775845-d7c55c80-ee79-11e9-94b9-490988c7bd54.png">
